src/PermissionServiceProvider.php

```diff
@@ -69,6 +69,16 @@ protected function registerBladeExtensions()
 
             $bladeCompiler->directive('hasanyrole', function ($arguments) {
                 list($roles, $guard) = explode(',', $arguments.',');
+                $roles = trim($roles);
```

Comment:
I've not followed it in detail, but my gut says that this could / should be solved in a much simpler way. Since the tests prove that this work, it should be relatively save to attempt to refactor this.

Also: this logic is being repeated for the other blade directive. If it can't be made simpler, create a new function so that logic appears only once in the codebase.